lib/MongoDB/_Server.pm

```diff
@@ -82,6 +83,24 @@ has is_master => (
     default => sub { {} },
 );
 
+has compressor => (
```

Comment:
As discussed above, maybe this should become an optional coderef that encapsulates the zlib_compression_level.  The closure generator could still live in _Protocol so all the  compression stuff is in one place and the `_build_compressor` method just calls `MongoDB::_Protocol::get_compressor($name, $opts)` or something like that.